packages/lesswrong/server/ckEditor/ckEditorWebhook.ts

```diff
@@ -99,20 +153,52 @@ async function handleCkEditorWebhook(message: any) {
     case "comment.removed":
     case "commentthread.removed":
     case "commentthread.restored":
+    case "collaboration.user.connected": {
+      const userConnectedPayload = payload as CkEditorUserConnectionChange;
+      const userId = userConnectedPayload?.user?.id;
+      const documentId = userConnectedPayload?.document?.id;
+
+      const result = await createMutator({
+        collection: CkEditorUserSessions,
+        document: {
+          userId,
+          documentId
+        },
+        validate: false,
+      });
+
+      break
+    }
     case "document.user.connected":
-      break;
+    case "collaboration.user.disconnected": {
+      const userDisconnectedPayload = payload as CkEditorUserConnectionChange;
+      const userId = userDisconnectedPayload?.user?.id;
+      const documentId = userDisconnectedPayload?.document?.id;
+      const userSession = await CkEditorUserSessions.findOne({userId, documentId }, {sort:{createdAt: -1}});
+
+      if (userSession) {
+        const result = await updateMutator({
+          collection: CkEditorUserSessions,
+          documentId: userSession._id, 
+          set: { endedAt: new Date(sent_at) },
+          validate: false,
+        });
+      }
+      break
+    }
     case "document.user.disconnected": {
-      interface CkEditorUserDisconnected {
+      interface CkEditorUserConnectionChange {
```

Comment:
If you're pulling this definition up to the top, I'd delete this one.  (Not even sure why it's letting you have two interfaces with the same name..)